src/stdlib/parse_auditd.rs

```diff
@@ -0,0 +1,435 @@
+    let bytes = bytes.try_bytes()?;
+    // check if bytes ends with newline, otherwise append it
+    // TODO: make the parser accept bytes without newline in the linux_audit_parser crate (to-be-contributed)
+    let bytes = if bytes.last() == Some(&b'\n') {
+        bytes
+    } else {
+        let mut bytes = BytesMut::from(bytes);
+        bytes.extend_from_slice(b"\n");
+        bytes.freeze()
+    };
+    let parsed = linux_audit_parser::parse(&bytes, false)?;
+
+    let mut log = ObjectMap::new();
+
+    let timestamp_millis = i64::try_from(parsed.id.timestamp)
+        .map_err(|_| Error::TimestampOverflow(parsed.id.timestamp))?;
+    // Should we use UTC as the timezone? The logs are generated with the system's
+    // timezone... What is the correct behavior? Maybe the system where vector is running
+    // have a different timezone than the system that generated the logs... so it is
+    // not correct to assume that the current system's timezone is the correct one
+    // (with TimeZone::timestamp_millis_opt)
+    // TODO: we should document that the timestamp is parsed into UTC timezone.
+    // TODO: Maybe we should accept in a parameter a custom timezone to parse the timestamp?
+    let Some(timestamp) = DateTime::from_timestamp_millis(timestamp_millis) else {
+        return Err(Error::TimestampOutOfRange(timestamp_millis).into());
+    };
+    log.insert("timestamp".into(), Value::from(timestamp));
+
+    let sequence = parsed.id.sequence;
+    log.insert("sequence".into(), Value::from(sequence));
+
+    if let Some(node) = parsed.node {
+        log.insert("node".into(), Value::from(Bytes::from(node)));
+    }
+
+    let message_type = parsed.ty.to_string();
+    log.insert("type".into(), Value::from(message_type));
+
+    let (enrichment, body): (ObjectMap, ObjectMap) = parsed
+        .body
+        .into_iter()
+        // TODO: improve this auditd crate with a IntoIter implementation for Body and not only
+        // for &Body, so we can have owned values. currently, the `into_iter` `Body` implementation
+        // relies on Vec::iter instead of Vec::into_iter, which forces us to clone the value in L56
+        .map(|(key, value)| {
+            let key = KeyString::from(key.to_string());
+            // TODO: remove this clone with a new auditd crate version (to-be-contributed yet)
+            let value = Value::try_from(value.clone());
+            value.map(|value| (key, value))
+        })
+        .collect::<Result<Vec<_>, _>>()?
+        .into_iter()
+        // Keys with only uppercase characters are considered enrichments of its lowercase counterpart
+        // https://github.com/linux-audit/audit-documentation/wiki/SPEC-Audit-Event-Enrichment
+        .partition(|(key, _)| key.chars().all(char::is_uppercase));
+
+    log.insert("body".into(), Value::from(body));
+
+    // TODO: should we downcase the keys of the enrichment so that they match its body counterparts?
+    if !enrichment.is_empty() {
```

Comment:
In auditd, the association between fields and its enriched value is done with capital letters ([see this documentation](https://github.com/linux-audit/audit-documentation/wiki/SPEC-Audit-Event-Enrichment)).

But that limitation is because the log format is plain and it seems like a workaround.

As we can have a more structured representation and the enrichment fields are separated, maybe we can downcase the keys and have something like

(deleted some object keys to simplify the example)
```json
{
    "body": {
        "auid": 4294967295,
        "uid": 0
    },
    "enrichment": {
        "auid": "unset",
        "uid": "root"
    }
}
```

so people can link the body fields and the enrichment fields easily, as this way, they enriched counterparts will exactly match the raw keys.

What do you think? I don't really know how people would really consume this so I would appreciate some feedback from people who would actually use this (https://github.com/vectordotdev/vrl/issues/66 requesters)